templates/workspace_services/azureml/terraform/roles.tf

```diff
@@ -20,7 +20,7 @@ data "azurerm_role_definition" "azure_ml_data_scientist" {
 }
 
 resource "azurerm_role_assignment" "app_role_members_aml_data_scientist" {
-  for_each           = toset(split("\n", data.external.app_role_members.result.principals))
+  for_each           = (data.external.app_role_members.result.principals == "") ? [] : toset(split("\n", data.external.app_role_members.result.principals))
```

Comment:
I haven't reviewed this (not sure what it's doing)

Reply:
I've tested it and works, if the principals string is empty gives an empty array so no role assignments are made. Previously it was trying to use an empty string as an object id. If there are principals, then works as before.